modules/seqexec/engine/src/main/scala/seqexec/engine/Execution.scala

```diff
@@ -85,6 +85,7 @@ object Execution {
     s => r match {
       case Result.OK(x)         => s.copy(runState = ActionState.Completed(x))
       case Result.OKStopped(x)  => s.copy(runState = ActionState.Completed(x))
+      case Result.OKAborted(x)  => s.copy(runState = ActionState.Completed(x))
```

Comment:
Here is your problem (I think). It should go to `ActionState.Aborted`. Then the next time you calculate the step state, you will get an aborted state.

Reply:
Hmm thanks, that seems to be the problem